### ISSUE TITLE
Scratch windows not always opening at the center of the screen when `center-new-windows` is true

### ISSUE THREAD

**Issue by madx** [paperwm/PaperWM#702]
**Describe the bug**
I am using [ULauncher](https://ulauncher.io/) as my application launcher/shortcut provider. It opens at random places of the screen depending on if I have other applications opened even though I've set it to always open on the scratch layer and I have `center-new-windows`  set to true in my Mutter configuration.

**To Reproduce**
1. Have no window on screen
2. Open ULauncher: it properly spawns in the center
3. Have a window opened
4. Open ULauncher: it appears in a corner of the screen

See attached video for a demo

**Expected behavior**
ULauncher always open in the center of the screen.

**Screenshots**
[Screencast from 2023-11-17 11-11-26.webm](https://github.com/paperwm/PaperWM/assets/5723/abe28c27-163f-42f4-9263-0cdc040d93b1)

**System information:**
Please execute `./gather-system-info.sh` in you PaperWM clone and paste the output below.

```
Distribution: Ubuntu
GNOME Shell 45.0-1ubuntu2
PaperWM branch/tag: Installed from EGO.
{
  "_generated": "Generated by SweetTooth, do not edit",
  "description": "Tiling window manager with a twist",
  "name": "PaperWM",
  "settings-schema": "org.gnome.shell.extensions.paperwm",
  "shell-version": [
    "45"
  ],
  "url": "https://github.com/paperwm/PaperWM",
  "uuid": "paperwm@paperwm.github.com",
  "version": 62,
  "version-name": "45.4.1"
}
Enabled extensions:
user-theme@gnome-shell-extensions.gcampax.github.com
gsconnect@andyholmes.github.io
one-thing@github.com
mprisLabel@moon-0xff.github.com
top-bar-organizer@julian.gse.jsts.xyz
do-not-disturb-while-screen-sharing-or-recording@marcinjahn.com
quick-settings-tweaks@qwreey
window-state-manager@kishorv06.github.io
paperwm@paperwm.github.com
auto-move-windows@gnome-shell-extensions.gcampax.github.com
pano@elhan.io
steal-my-focus-window@steal-my-focus-window
```

**Additional context**
To be honest I'm not sure if this is a PaperWM or a Mutter bug but I don't think I had a recent Mutter update and it was working properly last week

**Comment by jtaala** [comment]
Well, easy way to tell - disable PaperWM, logout, login and see if it still happens.

**Comment by jtaala** [comment]
Hmm - I can't reproduce this:

https://github.com/paperwm/PaperWM/assets/30424662/6cfe2758-e44b-446d-9390-0666ca931d8f

**Comment by madx** [comment]
> Well, easy way to tell - disable PaperWM, logout, login and see if it still happens.

Yup I was thinking about testing that, so I just did, and it works correctly:

[Screencast from 2023-11-17 12-13-36.webm](https://github.com/paperwm/PaperWM/assets/5723/05f08a22-456c-4d32-895d-376583377deb)

As soon as I reenable PaperWM the behavior is back.

Which version of ULauncher / Mutter are you using?

**Comment by madx** [comment]
I actually have the same issue with [Smile](https://smile.mijorus.it/) so I don't think this is a ULauncher issue

**Comment by jtaala** [comment]
Just double-checking you have ulauncher in winprops as scratch?

Can you disable your `center-new-windows` setting and retest?

**Comment by jtaala** [comment]
Ulauncher is v5.15.6
Mutter is 45.1-1

**Comment by jtaala** [comment]
ha - apparently I have center-new-windows enabled:

![image](https://github.com/paperwm/PaperWM/assets/30424662/f66f9fcb-2a53-489a-98ea-2ca11f3b2992)

**Comment by jtaala** [comment]
I'll keep trying to reproduce - if it's PaperWM and I can reproduce (and if it was working fine last week) then we'll be able to bisect and find the issue.

**Comment by jtaala** [comment]
Okay, one thing to check if PaperWM could be conflicting with another extension - can you try disable all other extensions (and just leave PaperWM enabled) and the logout/login again?

**Comment by madx** [comment]
Thanks for your help on this, I will try the different techniques and I'll also try bisecting with older versions of the extension!

**Comment by madx** [comment]
Alright, I did more investigation and I think we can mostly forget the above: it happens only when I have windows in the scratch layer. See the screencast below:

[Screencast from 2023-11-17 15-54-43.webm](https://github.com/paperwm/PaperWM/assets/5723/c3e05521-ab3d-4206-b81b-17688acaf544)

What I did change last week was that I have set KeepassXC (which uses an old and ugly Qt) as always on the scratch layer because it was messing around with the windowing (window was appearing on screens it wasn't supposed to be in)

I can work around this by not having it always opened and hopefully you can reproduce the behaviour with other apps.

I'm gonna make an uneducated guess at what could cause the issue: if the virtual screen of the scratch layer is wider than the physical screen, computations for the center could be messed up?

I will rename this issue accordingly as it is not specifically related to ULauncher.

**Comment by YounesJY** [comment]
@jtaala , @madx    — Hi there, did you find out how to resolve this?

I actually have the same issue with Ulauncher. It happens in two scenarios for me:

If I have an app in full-screen mode (e.g., F11 pressed), then Ulauncher opens in a random place near the top, not centered.

If there’s another app in the scratch layer, they seem to conflict, and Ulauncher ends up misplaced — again, not centered, but in a random position.

Let me know if you’ve found any workaround. Thanks!


https://github.com/user-attachments/assets/9d7b3c6d-b3ef-4ae4-8f81-12ad15ff3fe7

NOTE: i haven't done anything like "I have center-new-windows set to true in my Mutter configuration." 
but only "always open on the scratch layer" under PaperWM > WinProps

<img width="760" height="815" alt="Image" src="https://github.com/user-attachments/assets/62abf88a-36be-4782-8314-1c4886a2668b" />
